### ISSUE TITLE
Code hygiene: dedupe imports; remove unused; use zero_grad(set_to_none=True)

### ISSUE THREAD

**Issue by jashvira** [jashvira/policy-gradient-experiments#12]
Problem
- Duplicate/unused imports in grpo_train.py (Path, sys, scheduler builder).
- Uses optimizer.zero_grad() without set_to_none=True.

Impact
- Minor overhead and noise; small memory/throughput gains left on table.

Code
```python
# experiments/grpo/grpo_train.py
from pathlib import Path  # imported twice
import sys  # imported twice
from utils.optim_sched_utils import build_adamw, build_warmup_then_scheduler  # scheduler unused
...
optimizer.zero_grad()
```

Proposed fix
- Remove duplicate/unused imports; keep only what is used.
- Switch to optimizer.zero_grad(set_to_none=True).

Acceptance criteria
- No duplicate/unused imports in file.
- zero_grad uses set_to_none=True.
